### PR TITLE
Add room specfic approvers

### LINKED ISSUE #149: Split admin/approver permission
Admins should not necessarily be exposed to students to be chosen for booking. Split this list by adding an approvers list for each room that is independent of the admin role

### PR DESCRIPTION
**Pull request by logonoff**
Closes #149 #136

Changes the approver role so that each room has its own independent set of approvers that students must choose from.

This solves this issue of admins (who may not necessarily want to be approvers) from showing up as approvers when students booking rooms

### CODE REVIEW

File: frontend/src/theme/theme.tsx
Comment (huonggiangbui) [inline frontend/src/theme/theme.tsx:1]: Hmm we change the theme? any before/after demo?

File: frontend/src/pages/Room/joan6.tsx [hunk 1]
```diff
@@ -36,8 +38,9 @@ const EventsRow = ({ events }: { events: BookingRequest[] }) => {
                 ))}
                 <Box
                     sx={{
-                        top: `calc((100vh - 2em) / 14 * ${new Date().getHours() - 7})`,
-                        borderTop: `${theme.palette.app_colors.red} 4px solid`,
+                        top: `calc(((100vh - 5.25em) / 14) * ${new Date().getHours() - 7}${(new Date().getMinutes() / 60).toString().slice(1)})`,
```
Comment (huonggiangbui) [inline frontend/src/pages/Room/joan6.tsx:41]: what is this...

File: backend/src/api/routes/rooms.ts
Comment (ggggg) [inline backend/src/api/routes/rooms.ts:1]: are all changes in this file formatting changes?

File: backend/src/models/groupsModel.ts [hunk 2]
```diff
@@ -205,13 +207,14 @@ export default {
...
-    if (manager.role !== AccountRole.admin && !group.managers.some(x => x.utorid === manager.utorid)) {
+    // user must be either admin, a group manager. or trying to remove themselves
+    if (manager.role !== AccountRole.admin && !group.managers.some((x) => x.utorid === manager.utorid) && manager.utorid !== utorid) {
       return {
         status: 403,
         message: 'You are not allowed to modify this group.',
       };
     }
-    if (!group.members.some(x => x.utorid === utorid)) {
+    if (!group.members.some((x) => x.utorid === utorid)) {
```
Comment (huonggiangbui) [inline backend/src/models/groupsModel.ts:217]: we prolly need better variable names...

**Review by huonggiangbui — APPROVED** [review]
Left some recommended changes. The code quality is improving -- I'm glad !! Good job and keep going!

**Pull request merged**